src/main.js

```diff
@@ -2,25 +2,31 @@ import { Car } from "./domain/Car.js";
 import { CarValidator } from "./domain/CarValidator.js";
 import { Game } from "./Game.js";
 import { View } from "./view/View.js";
+import { assert } from "./utils/assert.js";
+import { isNumber } from "./utils/isNumber.js";
 
 const CAR_NAME_DELIMITER = ",";
 
 async function main() {
   try {
-    const input = await View.read("경주할 자동차 이름을 입력하세요.\n");
-    const carNames = input.split(CAR_NAME_DELIMITER);
+    const carNameInput = await View.read("경주할 자동차 이름을 입력하세요.\n");
+    const racingTimesInput = await View.read("시도할 회수는 몇회인가요?\n");
 
+    const carNames = carNameInput.split(CAR_NAME_DELIMITER);
     CarValidator.validateCarNames(carNames);
+    const racingTimes = Number(racingTimesInput);
+    assert(isNumber(racingTimes) && !Number.isNaN(racingTimes), "시도할 회수는 숫자값을 입력해주세요");
```

Comment:
숫자, 문자열 숫자인지 검증하는 간단한 로직이라 CarValidator와 같은 별도 모듈을 두지는 않고 유틸함수를 이용해서 검증해주었어요!

Reply:
isNaN이 아닌지에 대한 검증을 isNumber 내부에서 같이 할까 생각이 들기도 했는데 isNumber의 의미를 조금 해치는게 아닌가 싶어서 요렇게 밖에서 조건을 2개 잡아주었어요!

Reply:
말씀해주신 것처럼 `isNumber`가 여러 역할을 함께 하면 모호해질 수도 있겠다는 생각이 드는데요. 

`isNumber`의 역할이 typeof를 통해 타입을 검증하는 역할 밖에 없어 오히려 함수를 분리하지 않고 아래처럼 사용하는 것이 더 직관적으로 읽힐거 같다는 생각도 드는데 승완님 의견이 궁금합니다 !

```suggestion
    assert(typeof racingtimes === "number" && !Number.isNaN(racingTimes), "시도할 회수는 숫자값을 입력해주세요");
```

Reply:
혹은 `Number.isNaN()`까지 포함해서 `isValidRacingTImes` 같은 메서드로 분리를 해보는 것도 생각해 볼 수 있을거 같네용

Reply:
추가로 `assert`의 선언문을 볼 때는 잘 몰랐는데 이렇게 사용되는 곳을 보니 parameter를 객체로 받아 어떤 필드가 어떤 용도로 사용되는지 명시해줘도 좋겠다는 생각도 들었습니다 !

Reply:
> 말씀해주신 것처럼 isNumber가 여러 역할을 함께 하면 모호해질 수도 있겠다는 생각이 드는데요.
> 
> isNumber의 역할이 typeof를 통해 타입을 검증하는 역할 밖에 없어 오히려 함수를 분리하지 않고 아래처럼 사용하는 것이 더 직관적으로 읽힐거 같다는 생각도 드는데 승완님 의견이 궁금합니다 !

생각해보니까 저도 윤호님이 주신 의견에 동의하는 것 같아요! 여기 예시는 간단해서 딱 드러맞지는 않겠지만 이런게 저는 추상화가 내부 구현을 숨김으로서 코드를 이해하는데 있어서 조금 애매한 상황이 생길 수 있는 지점인 것 같아요.

의도한 방향은 "선언적으로 코드를 표현하겠다"와 같은 방향이었는데요.
저희가 얘기하고 있는 2가지 경우에서 각각 사람이 코드를 읽는다고 할 때는 아래와 같은 느낌일거라고 생각해요.
```
// 예를 들어 typeof racingTimes는 충분히 한번에 이해할 수 있는 단위이기는 하지만 약간 과장을 위해서 직독직해 하듯이 끊어서 읽어봤어요

assert(isNumber(racingTimes) && !Number.isNaN(racingTimes), "시도할 회수는 숫자값을 입력해주세요");
ㄴ (isNumber가 바로 보여서) 숫자인지 판단하는구나, racingTimes값이, 그리고 NaN은 아닌지

    assert(typeof racingTimes === "number" && !Number.isNaN(racingTimes), "시도할 회수는 숫자값을 입력해주세요");
ㄴ 데이터 타입을 볼건데, racingTimes가 숫자인지, 그리고 NaN은 아닌지
```

근데 저는 코드를 맨 왼쪽에서 읽을 때 바로 무엇을 하려고 하는지가 드러났으면 좋겠다는 의도에서 isNumber로 표현하고 싶었던 것 같아요. 근데 사실 이정도 구문은 개발 경험이 어느정도 있다면 끊어서 읽히는 것이 아니고 한번에 이해되는 맥락이며 너무 익숙하게 쓰는 코드 구문이다보니 오히려 이런 상황에서는 isNumber와 같은 추상화가 코드의 동작을 이해하는 데 한번 브레이크를 건다는 느낌이 드는 상황인 것 같아요. 추상화에 있어서 복잡도를 낮추고 재사용을 위한 목적을 포함하지 않고 오로지 함수의 의도를 드러내려는 목적으로만 추상화가 되는 이런? 상황은 오묘함이 있는 것 같아요! 고민해보니 스스로도 정리가 좀 되서 넘 좋네요! 👍

그래서 돌고돌아 저도 윤호님이 제안주신 방식이 더 직관적이라는 생각이 드네요!

Reply:
> 혹은 Number.isNaN()까지 포함해서 isValidRacingTImes 같은 메서드로 분리를 해보는 것도 생각해 볼 수 있을거 같네용

처음 남겨놓은 코멘트에는 데이터 검증 "조건"이 단순해서 일반적인 유틸함수로 검증한다고 남겨 놓았는데 생각해보니까 "조건"에 집중한다기 보다는 어떤 데이터인지를 생각해봤을때 도메인에 직접적인 데이터라서 저도 `isValidRacingTimes`와 같이 표현해주는 게 좋을 것 같아요. 

그 이유는 추상화 수준도 있고 도메인과 관련된 데이터 검증은 한 곳에 뭉쳐두는게 추후에 유지보수하는 입장에서 장점을 가져갈 수 있다라는 생각이 든 것 같아요!

요거는 기존 CarValidator 클래스에서 제공하는 형식에 맞게 GameValidator 요런거를 하나 두면 어떨까 싶어서 생각하다가 정적 메서드 하나를 두기에는 조금 과하다는 생각이 순간 들기도 했는데 Game과 관련된 validation은 이후에 추가될만 한 것 같고 데이터 검증을 하는 방식은 코드베이스 내에서 동일하게 가져가면 좋을 것 같아서 새로 하나 만들어보았어요! [4c08bc1](https://github.com/next-step/js-racingcar/pull/346/commits/4c08bc1f808f040ccf49724dd0aa2ac624f0095f)

Reply:
> 추가로 assert의 선언문을 볼 때는 잘 몰랐는데 이렇게 사용되는 곳을 보니 parameter를 객체로 받아 어떤 필드가 어떤 용도로 사용되는지 명시해줘도 좋겠다는 생각도 들었습니다 !

이 assert 함수를 저 스스로는 너무 자연스럽게 사용하고 있어서 생각을 못해봤었는데 해당 함수를 처음 보면 아무래도 직관적이진 않은 것 같아서 공감이 되네요!

근데 일반적으로 프로그래밍 언어들에서 제공하는 assert 함수에서 대부분 동일한 인터페이스를 갖추고 있어서 저는 크게 문제가 없다고 생각이 들기는 했어요! 저도 처음에는 잘 몰랐지만, 한번 알고난 이후에는 오히려 되게 직관적이라고 느껴지기도 해서 저는 만약 팀원이랑 얘기를 나누게 되었다면 처음엔 낯설 수 있지만 일반적으로 많이 사용하는 함수 시그니처이고, 한번만 구현부를 보면 된다고 생각해서 이런 관점은 어떻게 생각하시냐고 여쭈면서 한번 더 핑퐁을 해볼 것 같아요!

Reply:
>// 예를 들어 typeof racingTimes는 충분히 한번에 이해할 수 있는 단위이기는 하지만 약간 과장을 위해서 직독직해 하듯이 끊어서 읽어봤어요

저는 코드를 읽으면서 이런 생각을 했던거 같아요!

```
assert(isNumber(racingTimes) && !Number.isNaN(racingTimes), "시도할 회수는 숫자값을 입력해주세요");
// 숫자인지 확인하고 NaN이 아닌지 검사하는구나 -> 그런데 `isNumber()`는 타입만 확인해주는게 맞나 -> 함수를 확인해봐야겠다 -> 끝
```

그래서 오히려 추상화를 하지 않고 아래처럼 작성한다면 이렇게 읽힐거 같다는 생각을 했습니다 ㅎㅎ

```
 assert(typeof racingTimes === "number" && !Number.isNaN(racingTimes), "시도할 회수는 숫자값을 입력해주세요");

// 숫자인지 확인하고 NaN인지 검사하는구나 -> 끝
```

>요거는 기존 CarValidator 클래스에서 제공하는 형식에 맞게 GameValidator 요런거를 하나 두면 어떨까 싶어서 생각하다가 정적 메서드 하나를 두기에는 조금 과하다는 생각이 순간 들기도 했는데 Game과 관련된 validation은 이후에 추가될만 한 것 같고 데이터 검증을 하는 방식은 코드베이스 내에서 동일하게 가져가면 좋을 것 같아서 새로 하나 만들어보았어요! [4c08bc1](https://github.com/next-step/js-racingcar/pull/346/commits/4c08bc1f808f040ccf49724dd0aa2ac624f0095f)

👍 👍 👍 




>일반적으로 많이 사용하는 함수 시그니처이고, 한번만 구현부를 보면 된다고 생각해서 이런 관점은 어떻게 생각하시냐고 여쭈면서 한번 더 핑퐁을 해볼 것 같아요!

오 제 식견이 짧아 일반적으로 많이 사용하는 함수 시그니처인 것은 인지를 못 하고 있었네요 ㅎㅎ 그런 이유라면 충분히 납득이 가능할거 같습니다!